apps/mymedicare_cb/models.py

```diff
@@ -78,20 +99,64 @@ def create_beneficiary_record(username=None,
                               email="",
                               user_id_type="H"):
 
+    # Validate argument values. TODO: Replace asserts with exception handling.
+    if username is None:
+        mesg = "username can not be None"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
     assert username is not None
+
+    if username == "":
+        mesg = "username can not be an empty string"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
     assert username != ""
+
+    if user_hicn_hash is None:
+        mesg = "user_hicn_hash can not be None"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
+    else:
+        if len(user_hicn_hash) != 64:
+            mesg = "incorrect user HICN hash format"
+            log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
     assert user_hicn_hash is not None
     assert len(user_hicn_hash) == 64, "incorrect user HICN hash format"
+
+    # If mbi_hash is not NULL, perform length check.
+    if user_mbi_hash is not None:
+        if len(user_mbi_hash) != 64:
+            mesg = "incorrect user MBI hash format"
+            log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
+        assert len(user_mbi_hash) == 64, "incorrect user MBI hash format"
+
+    if fhir_id is None:
+        mesg = "fhir_id can not be None"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
     assert fhir_id is not None
+
+    if fhir_id == "":
+        mesg = "fhir_id can not be an empty string"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
     assert fhir_id != ""
 
     if User.objects.filter(username=username).exists():
-        raise ValidationError("user already exists", username)
+        mesg = "user already exists"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
+        raise ValidationError(mesg, username)
 
     if Crosswalk.objects.filter(_user_id_hash=user_hicn_hash).exists():
+        mesg = "user_hicn_hash already exists"
+        log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
         raise ValidationError("user_hicn_hash already exists", user_hicn_hash)
 
+    # If mbi_hash is not NULL, perform check for duplicate
+    if user_mbi_hash is not None:
+        if Crosswalk.objects.filter(_user_mbi_hash=user_mbi_hash).exists():
+            mesg = "user_mbi_hash already exists"
+            log_create_beneficiary_record(logger, username, fhir_id, user_mbi_hash, user_hicn_hash, mesg)
+            raise ValidationError("user_mbi_hash already exists", user_hicn_hash)
```

Comment:
Small style note: all of these `ValidationError`s can use the `mesg` variables in their scope, which would reduce the chance of copy-paste goofs for you.

Reply:
> Small style note: all of these `ValidationError`s can use the `mesg` variables in their scope, which would reduce the chance of copy-paste goofs for you.

@karlmdavis TY!  Missed that during edits. Updated the code for this. Looks much better!   Also, will convert all those ASSERTS to exceptions as part of another upcoming Jira story. Was tempted to do so now, but not wanting to change too much here.